File: PARCIAL LENGUAJES/PUNTO 4/euclides.c
#include <stdio.h>
#include <time.h>

int gcd(int a, int b){
    if(b == 0)
        return a;
    return gcd(b, a % b);
}

int main(){

    int a = 987654321;
    int b = 123456789;

    clock_t inicio = clock();

    int resultado = gcd(a,b);

    clock_t fin = clock();

    double tiempo = (double)(fin - inicio) / CLOCKS_PER_SEC;

    printf("C (compilado)\n");
    printf("MCD(%d,%d) = %d\n", a,b,resultado);
    printf("Tiempo: %f segundos\n", tiempo);

    return 0;
}

File: PARCIAL LENGUAJES/PUNTO 4/euclides.hs
import System.CPUTime

gcdE :: Integer -> Integer -> Integer
gcdE a 0 = a
gcdE a b = gcdE b (mod a b)

main = do
    start <- getCPUTime
    
    let a = 987654321
    let b = 123456789
    let resultado = gcdE a b

    end <- getCPUTime

    let tiempo = fromIntegral (end - start) / (10^12)

    putStrLn "Haskell (declarativo)"
    putStrLn ("MCD(" ++ show a ++ "," ++ show b ++ ") = " ++ show resultado)
    putStrLn ("Tiempo: " ++ show tiempo ++ " segundos")
